src/System.Runtime.Numerics/tests/ComplexTests.cs

```diff
@@ -226,11 +226,32 @@ public static IEnumerable<object[]> Abs_Advanced_TestData()
                 foreach (double invalidImaginary in s_invalidDoubleValues)
                 {
                     yield return new object[] { RandomPositiveDouble(), invalidImaginary, Math.Abs(invalidImaginary) }; // Invalid imaginary
-                    yield return new object[] { invalidReal, invalidImaginary, (double.IsInfinity(invalidReal) || double.IsInfinity(invalidImaginary)) ? double.PositiveInfinity : double.NaN }; // Invalid real, invalid imaginary
+                    yield return new object[] { invalidReal, invalidImaginary, (double.IsNaN(invalidReal) || double.IsNaN(invalidImaginary)) ? double.NaN : double.PositiveInfinity }; // Invalid real, invalid imaginary
```

Comment:
The current version looks correct to me. You don't need to do anything about `NetcoreUwp` because this implementation should be shared with that platform, so the "advanced" tests will work there.